.github/workflows/frontend_ci.yml

```diff
@@ -23,6 +23,8 @@ jobs:
   pre-job:
     name: Pre Job
     runs-on: ubuntu-latest
+    permissions:
```

Comment:
I believe we already have read/write (the default for [permissive](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) mode), so you are reducing permissions here, rather than adding.